### PR TITLE
[form-builder] Migrate Details component to Sanity UI

### PR DESCRIPTION
**Pull request by mariuslundgard**
The `Details` component is used by:

- `inputs/Warning.tsx`
- `inputs/arrays/ArrayOfObjectsInput.tsx`
- `inputs/ObjectInput/UnknownFields.tsx`

### CODE REVIEW

File: packages/@sanity/form-builder/src/components/Details.tsx [hunk 1]
```diff
@@ -1,50 +1,71 @@
...
+`
 
-export class Details extends React.Component<DetailsProps, DetailsState> {
-  static defaultProps = {
-    title: 'Details',
-    isOpen: false,
-  }
+const Header = styled(Flex)`
+  cursor: default;
+  user-select: none;
```
Comment (bjoerge) [inline packages/@sanity/form-builder/src/components/Details.tsx:27]: What's the reason for `user-select: none` here? 

It seems to be different from how [native `<details>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) elements work.

File: packages/@sanity/form-builder/src/components/Details.tsx [hunk 2]
```diff
@@ -1,50 +1,71 @@
...
-          </div>
-        </button>
-        <div className={styles.content}>{children}</div>
-      </div>
-    )
-  }
+export function Details(props: DetailsProps) {
+  const {children, open: openProp, title = 'Details'} = props
+
+  const [open, setOpen] = useState(openProp || false)
```
Comment (bjoerge) [inline packages/@sanity/form-builder/src/components/Details.tsx:44]: Can we assume it's controlled by the consumer If we receive a non-undefined `open` prop and not maintain internal state here? As it is now a consumer can't fully control its open/closed state.

**Review by bjoerge — COMMENTED** [review]
Looks good, left a couple of comments.

**Review by benedicteb — APPROVED** [review]

File: packages/@sanity/form-builder/src/components/Details.tsx (hunk 2 above)
Comment (mariuslundgard) [reply packages/@sanity/form-builder/src/components/Details.tsx:44]: Fixed

**Pull request merged**